spec/blockchain/blockchain.md

```diff
@@ -377,10 +377,13 @@ The first block has `block.Header.AppHash == []byte{}`.
 ### LastResultsHash
 
 ```go
-block.ResultsHash == MerkleRoot(state.LastResults)
+block.LastResultsHash == MerkleRoot(ResponseBeginBlock.Events, MerkleRoot([]ResponseDeliverTx), ResponseEndBlock.Events)
 ```
 
-MerkleRoot of the results of the transactions in the previous block.
+`LastResultsHash` is the root hash of a Merkle tree w/ 3 leafs: amino-encoded
+`ResponseBeginBlock#Events`, root hash of a Merkle tree build from
+`ResponseDeliverTx` responses (Log, Info and Codespace fields are ignored), and
+amino-encoded `ResponseEndBlock#Events`.
```

Comment:
The references to Amino here and elsewhere probably won't be valid once we migrate to Protobuf?